Vishnu/frontend

vvijay874 opened the pull request:
Created a frontend for the website, including:
- A home page
- A footer
- A navbar

vvijay874 committed:
Set up tailwindcss

Finished tailwindcss setup!

vvijay874 committed:
Created footer, navbar, and home page

Created the footer, navbar, and home page of the website. These have working links that link to the other pages. This is a basic framework for how the website is going to look in the future

vvijay874 committed:
Updated some colors

Changed purple (outdated) to emerald, which looks nicer and works better

MichaelDeng03 commented on client/src/output.css:
Add this to your gitignore. It can be rebuilt using your tailwind config

MichaelDeng03 commented on client/src/pages/Home.js:
```diff
@@ -0,0 +1,45 @@
+              <div className="flex flex-col justify-center">
+                <h3 className="text-3xl font-bold mb-4 text-emerald-600">INSERT INFORMATION HERE</h3>
+                <p className="text-lg mb-4">
+                    INSERT INFORMATION HERE
```
What type of information may belong here? 
I'm alright with you leave placeholders, but only if you the category that they'll end up being.

MichaelDeng03 replied:
As an aside, you might find it nice to use TODO comments, or otherwise a consistent templating pattern. This will help you keep track of all the placeholders that need to be filled out by just search and replacing. Otherwise, it's easy for a few to slip through on accident.

MichaelDeng03 reviewed (changes requested):
I'm getting an Axios error trying to run this

```
cd client
npm i
npm start
```
```
Compiled with problems:
ERROR in ./src/App.js 8:0-26
Module not found: Error: Can't resolve 'axios' in '/home/madeng/AbstractOperator/AbstractOperatorsInterns/client/src'
ERROR in ./src/pages/FileUpload.js 6:0-26
Module not found: Error: Can't resolve 'axios' in '/home/madeng/AbstractOperator/AbstractOperatorsInterns/client/src/pages'
ERROR in ./src/pages/SocialBot.js 6:0-26
Module not found: Error: Can't resolve 'axios' in '/home/madeng/AbstractOperator/AbstractOperatorsInterns/client
```

Could you also create a README.md on all the steps required to run this client?
Imagine you're trying to communicate with somebody who is technical, but not experienced with react or node.

vvijay874 committed:
Added video to frontend

Added a video to the frontend and a triangle to separate it from the rest of the page

vvijay874 committed:
Added overlay text

Added overlay text on top of the video

vvijay874 committed:
Added a little phone image

Added a phone image (to be changed in the future)

vvijay874 committed:
Finished NavBar

Edited the navbar so that it looks more professional

vvijay874 committed:
Triangle is now responsive

The triangle now covers all of the video no matter the screen size!

vvijay874 committed:
Created SocialBot interface and Log in

SocialBot page now has a basic UI to work with, and the login pulls up a dropdown. However, the dropdown is above the button, which is an error that will be fixed

vvijay874 committed:
Fixed dropdown error

The dropdowns for signing up and logging in now appear below the buttons as expected.

vvijay874 committed:
Tried fixing multiple dropdown error

Multiple dropdowns can show up at the same time, which blocks some output. I tried fixing this, but it needs boolean algebra that is not working right now

vvijay874 committed:
Updated bugs

- Fixed drop downs showing up conditionally
- Fixed social situation not showing up once signed in
